File: app/backend/src/tv_watchlist/models/category_summary.py
"""Sidebar-level view of a category."""

from __future__ import annotations

from pydantic import BaseModel

from tv_watchlist.models.watch_counts import WatchCounts


class CategorySummary(BaseModel):
    """Everything the category list needs, without the row payload."""

    id: str
    name: str
    file_name: str
    sheet_title: str
    accent: str
    mtime: float
    has_watch_column: bool
    locked_by_excel: bool
    hero_url: str | None = None
    counts: WatchCounts

File: app/frontend/src/types/CategorySummary.ts
import type { WatchCounts } from "./WatchCounts";

export interface CategorySummary {
  id: string;
  name: string;
  file_name: string;
  sheet_title: string;
  accent: string;
  mtime: number;
  has_watch_column: boolean;
  locked_by_excel: boolean;
  hero_url?: string | null;
  counts: WatchCounts;
}
